crates/services/gas_price_service/src/lib.rs

```diff
@@ -152,21 +186,43 @@ where
                 tracing::debug!("Stopping gas price service");
                 should_continue = false;
             }
-            new_algo = self.update_algorithm.next() => {
-                let new_algo = new_algo?;
+            da_block_costs = self.da_block_costs_provider.recv() => {
+                self.da_block_costs = match da_block_costs {
+                    Ok(da_block_costs) => Some(da_block_costs),
+                    Err(err) => {
+                        tracing::error!("Failed to get da block costs: {:?}", err);
+                        None
```

Comment:
If you have some value already, but the next request failed, you will set it to be `None`, while old value still is good=)

Reply:
addressed in f0bbf4974